### PR TITLE
Doms maintenance update

### PR DESCRIPTION
**Pull request by EmCath**
Main Changes: 
 * Update java version (1.8 -> 11)
  * Comply with forbidden api
  * Update junit (4 -> 5), with corresponding changes to tests
  * Update most external dependencies

### CODE REVIEW

File: src/main/java/dk/statsbiblioteket/larm_doms_exporter/persistence/dao/DomsExportRecordDAO.java [hunk 1]
```diff
@@ -20,7 +20,7 @@ public DomsExportRecordDAO(HibernateUtilIF hibernateUtilIF) {
      * @return the timestamp.
      */
     public Long getMostRecentExportedTimestamp() {
-       List<DomsExportRecord> records = getSession().createQuery("FROM DomsExportRecord WHERE state = ? ORDER BY lastDomsTimestamp DESC ").setParameter(0, ExportStateEnum.COMPLETE).list();
+       List<DomsExportRecord> records = getSession().createQuery("FROM DomsExportRecord WHERE state = ?0 ORDER BY lastDomsTimestamp DESC ").setParameter(0, ExportStateEnum.COMPLETE).list();
```
Comment (HalinaHolst) [inline src/main/java/dk/statsbiblioteket/larm_doms_exporter/persistence/dao/DomsExportRecordDAO.java:23]: state = ?0 - er det en ny syntaks, kan se det samme længere nede? Hvad betyder det?

Comment (EmCath) [reply src/main/java/dk/statsbiblioteket/larm_doms_exporter/persistence/dao/DomsExportRecordDAO.java:23]: Det skulle meget gerne betyde det samme, "?" er blevet legacy syntax og vi får kastet en "java.lang.IllegalArgumentException: org.hibernate.QueryException: Legacy-style query parameters (`?`) are no longer supported; use JPA-style ordinal parameters (e.g., `?1`) instead"

File: pom.xml [hunk 2]
```diff
@@ -21,52 +21,41 @@
...
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
-            <version>1.6.4</version>
+            <version>1.7.9</version>
         </dependency>
         <dependency>
             <groupId>log4j</groupId>
             <artifactId>log4j</artifactId>
-            <version>1.2.16</version>
+            <version>1.2.17</version>
```
Comment (kingudk) [inline pom.xml]: There should only be one log implemtation used (logback should be preffered)

File: pom.xml [hunk 3]
```diff
@@ -75,24 +64,24 @@
...
             <groupId>commons-cli</groupId>
             <artifactId>commons-cli</artifactId>
-            <version>1.1</version>
+            <version>1.4</version>
         </dependency>
         <dependency>
             <groupId>postgresql</groupId>
             <artifactId>postgresql</artifactId>
-            <version>8.4-701.jdbc4</version>
+            <version>9.2-1002.jdbc4</version>
```
Comment (kingudk) [inline pom.xml]: Postgresql driver should updated to a much newer version. Try to see if 
<dependency>
    <groupId>org.postgresql</groupId>
    <artifactId>postgresql</artifactId>
    <version>42.2.19</version>
</dependency>
works as expected

File: src/main/java/dk/statsbiblioteket/larm_doms_exporter/consumer/processors/DoExportProcessor.java [hunk 4]
```diff
@@ -130,7 +131,7 @@ protected void processThis(DomsExportRecord record, ExportContext context, Expor
         logger.info("Writing new export file " + outputFile.getAbsolutePath());
         try {
             FileOutputStream os = new FileOutputStream(tempOutputFile);
-            os.write(result.getBytes());
+            os.write(result.getBytes("utf-8"));
```
Comment (kingudk) [inline src/main/java/dk/statsbiblioteket/larm_doms_exporter/consumer/processors/DoExportProcessor.java]: Consider using StandardsCharset.UTF8.name()

File: pom.xml [hunk 5]
```diff
@@ -21,52 +21,41 @@
...
-            <version>3.3.1.GA</version>
+            <version>5.4.15.Final</version>
         </dependency>
         <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <version>4.9</version>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.6.2</version>
```
Comment (kingudk) [inline pom.xml]: Time flies and junit can be updated to 5.7.1 :)

File: Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+
+openshift.withCluster() { // Use "default" cluster or fallback to OpenShift cluster detection
+
+
+    echo "Hello from the project running Jenkins: ${openshift.project()}"
+
+    //Create template with maven settings.xml, so we have credentials for nexus
+    podTemplate(
+            inheritFrom: 'kb-jenkins-agent-java-11',
```
Comment (kingudk) [inline Jenkinsfile]: Should be changed to: "inheritFrom: 'maven'," as we now use the standard maven agent image.

File: pom.xml [hunk 7]
```diff
@@ -21,52 +21,41 @@
...
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-api</artifactId>
-            <version>1.6.4</version>
+            <version>1.7.9</version>
```
Comment (kingudk) [inline pom.xml]: should be 1.7.30 instead

**Review by kingudk — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 5 above)
Comment (EmCath) [reply pom.xml]: When I update junit, the tests fail to run (0 tests get run)

Comment (kingudk) [reply pom.xml]: fixed by updating parentpom

**Review by kingudk — APPROVED** [review]

**Pull request merged**